packages/block-library/src/separator/edit.js

```diff
@@ -6,31 +6,66 @@ import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
+import { useEffect } from '@wordpress/element';
 import { HorizontalRule } from '@wordpress/components';
-import { withColors, useBlockProps } from '@wordpress/block-editor';
-/**
- * Internal dependencies
- */
-import SeparatorSettings from './separator-settings';
+import {
+	useBlockProps,
+	getColorClassName,
+	__experimentalUseColorProps as useColorProps,
+} from '@wordpress/block-editor';
+import { usePrevious } from '@wordpress/compose';
+
+export default function SeparatorEdit( { attributes, setAttributes } ) {
+	const { backgroundColor, style, opacity } = attributes;
+	const color = useColorProps( attributes );
+
+	// V1 blocks expect to have a default opacity of 0.4, so on transformation
+	// get an opacity attribute set to css in order to assign a class name
+	// to provide this opactiy. Once the user changes the color this is removed
+	// and opacity can then be set via the color block support alpha channel.
+
+	useEffect( () => {
```

Comment:
If we add a default value for `opacity` in `block.json`, could that avoid requiring the `useEffect` here? I wasn't sure the `useEffect` is required for transformation. Apologies if I'm missing context from [your comment](https://github.com/WordPress/gutenberg/pull/38428#issuecomment-1035858501) about adding in the new attribute!

Reply:
Related to the comment above - because a default attribute doesn't serialise the only way  I could see to differentiate with v1 blocks with no attributes was to set this manually, ie. if there is no opacity attribute it must be a new block as all v1 blocks would have gone through a transformation and had one added.